executor.go

```diff
@@ -48,9 +52,190 @@ type ValidationResult struct {
 }
 
 // Executor is defined to validate artifacts.
-type Executor struct{}
+type Executor struct {
+	// Executor could use multiple verifiers to validate artifacts. Required.
+	Verifiers []Verifier
+
+	// Executor should configure exactly one store to fetch supply chain
+	// content. Required.
+	Store Store
+
+	// Executor should have at most one policy enforcer to evalute reports. If
+	// not set, the validation result will be returned without evaluation.
+	// Optional.
+	PolicyEnforcer PolicyEnforcer
+}
+
+// NewExecutor creates a new executor with the given verifiers, store, and
+// policy enforcer.
+func NewExecutor(verifiers []Verifier, store Store, policyEnforcer PolicyEnforcer) (*Executor, error) {
```

Comment:
Since `PolicyEnforcer` is optional, we can remove it from the parameter.
```suggestion
func NewExecutor(store Store, verifiers []Verifier) (*Executor, error) {
```